modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gmos/GmosControllerEpics.scala

```diff
@@ -109,39 +115,50 @@ class GmosControllerEpics[T<:GmosController.SiteDependentTypes](encoders: GmosCo
     _ <- DC.setCcdYBinning(encode(dc.bi.y))
   } yield ()
 
-
+  private def smartSetParam[A: Equal](v: A, get: => Option[A], set: SeqAction[Unit]): SeqAction[Unit] =
+    get.flatMap(g => (g =/= v).option(set)).getOrElse(SeqAction.void)
 
   def setFilters(f: T#Filter): SeqAction[Unit] = {
     val (filter1, filter2) = encoders.filter.encode(f)
 
-    CC.setFilter1(filter1) *> CC.setFilter2(filter2)
+    smartSetParam(filter1, GmosEpics.instance.filter1, CC.setFilter1(filter1)) *>
+      smartSetParam(filter2, GmosEpics.instance.filter2, CC.setFilter2(filter2))
   }
 
   @SuppressWarnings(Array("org.wartremover.warts.Equals"))
   def setDisperser(d: GmosController.Config[T]#GmosDisperser): SeqAction[Unit] = {
     // TODO: add support for Enum parameters in acm, and then define Enum type for disperserMode
-    val disperserMode = "WLEN"
-    CC.setDisperser(encoders.disperser.encode(d.disperser)) *>
-      CC.setDisperserMode(disperserMode) *>
-      d.order.filter(_ => d.disperser != Disperser.MIRROR).fold(SeqAction.void)(o => CC.setDisperserOrder(encode(o))) *>
-      d.lambda.filter(_ => d.disperser != Disperser.MIRROR && !d.order.contains(Order.ZERO)).fold(SeqAction.void)(o => CC.setDisperserLambda(encode(o)))
+    val disperserMode0 = "WLEN"
+    val disperserMode1 = "SEL"
+    val disperser = encoders.disperser.encode(d.disperser)
+    def disperserModeDecode(v : Int): String = if(v===0) disperserMode0 else disperserMode1
+
+    smartSetParam(disperser, GmosEpics.instance.disperser, CC.setDisperser(disperser)) *>
```

Comment:
This is a bit hard to follow, could it be perhaps split into several functions. Up to you